content/strategy-goals/strategy/code-graph/search/product.md

```diff
@@ -10,37 +10,71 @@ Quicklinks:
+
+- Implementing streaming across backends to minimize latency and memory usage
+- Partitioning search code into different domains and stages to manage complexity
+- Simplifying and unifying search result types
+
+The primary goal of these efforts has been to push the backend to a state where it will be maintainable in the face of new features. This is an ongoing process, but we’ve gained significant ground.
+
+Q2's work will start to shift the focus of backend work from maintainability to capability and flexibility. This means thinking of “search” more as an API that we provide for internal consumers that exposes all the information needed to build features on top of it. Though search is an API, it's not yet flexible enough to solve all the use cases presented by, for example, Code Insights, and some other proposed user-facing query features. This work is important because it enables development of high-value roadmap items, and the ability to conceive of novel features that weren't possible before.
+
+#### Search UX
+
+In light of new search capabilities, we believe there's a big opportunity to create an outstanding search experience, which is vital to Sourcegraph's continued dominance in the code search market.
+
+This foundational work will have design and technical components. Design will focus on digesting user feedback, conducting user research, and iterating on UI and UX solutions. Engineering will focus on enabling and collaborating with design to determine the execution path, specify a development roadmap, and start development.
+
+### Research
+
+Our research efforts will consist of establishing a baseline for UX issues by conducting a UX audit and building an estimated and prioritized design debt and UX roadmap. We will also conduct user research on Notebooks usage to inform its future roadmap.
+
+Despite being research, there will be some delivery related to this work. We plan to release an incremental improvement to the logged-in homepage, refactor the search results UI components, and execute on technical work related to our migration off of the Monaco editor.
 
-Sourcegraph needs to continue to differentiate its core platform beyond code search. Search will always power our platform, but to remain competitive we need to become more than the world's best code search. We must provide compelling solutions for the most pressing use cases of our customers. Our goal is to ensure the base Sourcegraph platform (currently Search, Notebooks, and Code monitors) continues to become more compelling and differentiated and supports future innovation from our team and across Sourcegraph.
+We're also making a bet on natural language search, exploring how we might incorporate ML models to generate query suggestions from user input. This is an interesting approach because, rather than training a model to semantically classify code, we'll be training it to generate a valid Sourcegraph query from natural language input.
```

Comment:
Making a bet to me sounds more like we've already decided to invest in it. I realize it's in the research section, but I'd change this to something more like we're going to start exploring, rather than that we are placing a bet on it.

Reply:
```suggestion
We'll also start to explore usage of ML models to enable natural language search, whether through generating query suggestions from natural language, or directly building a natural language search backend. We'll evaluate both approaches, and spike out the work required to get them in front of users. We expect this to be a long-term effort, and as such we are not targeting actual integration into the Sourcegraph product this quarter.
```

Reworded based on @jyavorska's comment, also clarifying that there are two paths we can potentially take (and we'll likely evaluate both), and what we expect to achieve (and not achieve) this quarter.

I agree we can remove the specifics of how this might work below.